Jenkinsfile

```diff
@@ -14,90 +32,48 @@ pipeline {
     parameters {
         string(name: "REMOTE_SWF_LINK", defaultValue: "https://github.com/Pelagicore/software-factory.git")
         string(name: "BRANCH_SWF", defaultValue: "master")
-        string(name: "SWF_DIR", defaultValue: "SWF_SRC")
-        string(name: "SWF_BP_DIR", defaultValue: "SWF_BP_SRC")
     }
 
     stages {
         stage('Download') {
             steps {
-
                 // Delete old files
                 sh 'rm -rf .[^.] .??* *'
 
-                // Make separate directories for SWF and SWF-BP for testing if the changeset is compatible with both.
-                sh "mkdir ${SWF_DIR}"
-                sh "mkdir ${SWF_BP_DIR}"
-
-                // Get The latest SWF src
-                dir ("${SWF_DIR}") {
-                    retry(2) {
-                        sh "git init"
-                    }
-
-                    sh "git remote add jenkins ${REMOTE_SWF_LINK}"
-                    sh "git fetch jenkins ${BRANCH_SWF}"
-                    sh "git checkout jenkins/${BRANCH_SWF}"
-
-                    // Initialize the submodules and fetch the new code change into it
-                    sh "git submodule init && git submodule update"
-                }
+                // Get the latest SWF src
+                sh "git clone ${REMOTE_SWF_LINK} ${SWF_DIR}"
+                sh "cd ${SWF_DIR} && git checkout origin/${BRANCH_SWF}"
 
                 // Checkout BP src along wth changes in pull request
+                sh "mkdir ${SWF_BP_DIR}"
                 dir ("${SWF_BP_DIR}") {
                     checkout scm
                 }
 
                 // Copying SWF BP code to SWF build, faking a "bump"
-                sh "cp -r ${SWF_BP_DIR}/* ${SWF_DIR}/docs/swf-blueprint/" // It's for testing (BP changes + SWF code).
-            }
-        }
-
-        // Configure and build SWF BP
-        stage('Configure SWF-BP') {
-            steps {
-                script {
-                    String buildParams = "-DENABLE_PDF=OFF"
-                    sh "cd ${SWF_BP_DIR} && cmake -H. -Bbuild ${buildParams}" // Configure for SWF BP
-                }
-            }
-        }
-
-        stage('Build SWF-BP') {
-            steps {
-                sh "cd ${SWF_BP_DIR}/build && make" // Build SWF-BP code.
-            }
-        }
-
-        // Configure and build SWF
-        stage('Configure SWF') {
-            steps {
-                script {
-                    String buildParams = "-DENABLE_PDF=OFF"
-                    sh "cd ${SWF_DIR} && cmake -H. -Bbuild ${buildParams}" // Configure for SWF baseline
-                }
+                // It's for testing (BP changes + SWF code)
+                sh "cp -r ${SWF_BP_DIR}/* ${SWF_DIR}/docs/swf-blueprint/"
```

Comment:
Just wanted to check if this builds the swf-blueprint as well along with SWF code whenever you lunch build from SWF. Because there is no submodule init done here. Can you clarify on this ?

Reply:
Yes, the `${SWF_DIR}/docs/swf-blueprint/` directory is empty until one initializes the submodule. Since we don't want the actual submodule, but rather the one from whatever version of the blueprint we're building, we don't need to do `init` but we can just copy the files that we already have from our `checkout scm` step.